src-docs/src/components/guide_section/guide_section.tsx

```diff
@@ -40,6 +40,10 @@ export const GuideSectionCodeTypesMap = {
     name: 'demoJS',
     displayName: 'Demo JS',
   },
+  TSX: {
+    name: 'demoTSX',
+    displayName: 'Demo JS',
+  },
```

Comment:
@cchaos recommended leaving the tab name for tsx demo's the same as JS - I have no strong opinions either way here, def feel to chime in if you have thoughts

Reply:
I'd be in favor of changing this to Demo TS (or spelled out TypeScript), but I'm not sure as it could make the demo experiences feel more fragmented while the examples are converted over. Anyone else have thoughts on this?

Reply:
I didn't think it felt necessary to discern since _**most**_ examples don't have actual TS in them and so works in either a purely JS or TS environment. But TS is just a typed JS system that's why I opted for leaving JS. The other option is to leave out the language completely and find a name that exemplifies demo code vs snippet

Reply:
It might be worth distinguishing `TS` code for users who aren't familiar with Typescript and copy and paste an entire example code block into their `.js`/`.jsx` files, and are then confused as to why they get syntax errors. That's not a great dev user experience, and it would be good to provide some affordance/hint that the code we're providing won't work OOTB in plain JS.

Reply:
On the other hand it could also scare some people off if it doesn't actually have Typescript in the demo code.

Reply:
Right, but since we can deliberately specify `.TSX` in the passed source type, we can deliberately only change the tab copy for demo code that actually has Typescript (which would be exactly what's happening in this PR where only 1 demo is changing).

Reply:
So do you think it matters to specifically say `Demo TS` when the demo example might be a `.TSX` file but doesn't actually contain anything specific to TypeScript?

Reply:
I think it's fine to avoid changing the type to `.TSX` and leave it as `.JS` for files that don't actually have any Typescript/typing in them (even if the demo file is a `.tsx`). I primarily want `.TSX` to be specified on demo code that will error in normal `.js` files.

If in the some imaginary future we convert _all_ our demos/docs files to `.tsx`, it might make sense to do a wholesale change at that point and just generally communicate that all our code expects Typescript - that's probably getting ahead of the though 🏇

Reply:
Ok, SGTM

Reply:
Sweet! Anyone have any thoughts on `Demo TS` vs. spelled out `Demo Typescript`? 🤔 

```suggestion
  TSX: {
    name: 'demoTSX',
    displayName: 'Demo TS',
  },
```

```suggestion
  TSX: {
    name: 'demoTSX',
    displayName: 'Demo Typescript',
  },
```

I'm leaning _maybe_ Typescript?... but no strong feelings either way personally

Reply:
Let's just go `Demo TS` for now because I still think it's important to note that it's the code that makes up the demo and not the code that makes up the component.

Reply:
Also, space... 😆